src/clj/garden/color.clj

```diff
@@ -330,6 +470,168 @@
         b (Math/round ^Double (* 0xff (+ b' m)))]
     (Rgb. r g b)))
 
+(def ^:private lab-constants
+  " Note: xyz-lab will be called f(t) below for brevity and consistency with literature.
+
+  xn, yn, zn - CIE XYZ tristimulus values for the white point under Illuminant D65.
+  (See https://en.wikipedia.org/wiki/Lab_color_space#CIELAB-CIEXYZ_conversions)
+
+  t3 - The point at which f(t) is made discontinuous.
+  Because f(t) approaches infinity at t=0, f(t) is made linear below below this value.
+  An approximation of (6/29^3).
+
+  t2 - The slope of the linear part of f(t).
+  An approximation of the inverse of (1/3)((29/6)^2).
+
+  t1 - The slope of the reverse transformation of f(t).
+  An approximation of 3(6/29)^2.
+
+  t0 - The intercept of the linear part of f(t).
+  An approximation of 4/29."
+  {:xn 0.950470
+   :yn 1
+   :zn 1.088830
+   :t0 0.137931034
+   :t1 0.206896552
+   :t2 0.12841855
+   :t3 0.008856452})
+
+(defn- rgb->xyz
+  "Intermediate conversion from rgb to CIEXYZ"
+  [r g b]
+  (let [/ clj//
+        + clj/+
+        * clj/*
+        - clj/-
+        rgb-xyz (fn [r]
+                  (let [r (/ r 255)]
+                    (if (<= r 0.04045)
+                      (/ r 12.92)
+                      (Math/pow (/ (+ r 0.055) 1.055) 2.4))))
+        xyz-lab (fn [t]
+                  (let [{:keys [t0 t2 t3]} lab-constants]
+                    (if (> t t3)
+                      (Math/pow t (/ 1 3))
+                      (+ t0 (/ t t2)))))
+        rgb (mapv rgb-xyz [r g b])
+        {:keys [xn yn zn]} lab-constants]
+    (mapv
+      (fn [consts n]
+        (xyz-lab (/ (apply + (map * consts rgb)) n)))
+      [[0.4124564 0.3575761 0.1804375]
+       [0.2126729 0.7151522 0.0721750]
+       [0.0193339 0.1191920 0.9503041]]
+      [xn yn zn])))
+
+(defn ^Lab rgb->lab
```

Comment:
It's probably worthwhile to mark these and the other helper functions as private as you have done above.